### PR TITLE
Making navigation target is optional for non-containment navigation property

### PR DESCRIPTION
**Pull request by sabyabr**
Allowing non-containment navigation property to not required to have navigation target set
https://github.com/OData/odata.net/issues/273

### CODE REVIEW

File: test/FunctionalTests/Tests/DataOData/Tests/OData.Query.TDD.Tests/Semantic/Functional/PathFunctionalTests.cs [hunk 1]
```diff
@@ -570,9 +570,8 @@ public void NavigationPropertyWithMissingEntitySetShouldThrow()
             var model = ModelBuildingHelpers.GetModelWithNavPropWithNoTargetSet();
```
Comment (congysu) [inline test/FunctionalTests/Tests/DataOData/Tests/OData.Query.TDD.Tests/Semantic/Functional/PathFunctionalTests.cs]: Also change the method name `NavigationPropertyWithMissingEntitySetShouldThrow`.

**Comment by lewischeng-ms** [comment]
Hi @sabyabr Thanks very much for your contribution. But we may need to have some discussion here. According to the [OData spec](http://docs.oasis-open.org/odata/odata/v4.0/odata-v4.0-part3-csdl.html):

7.1.3 Attribute Nullable

The edm:NavigationProperty element MAY contain the Nullable attribute whose Boolean value     specifies whether a navigation target is required for the navigation property.
If no value is specified for a navigation property whose Type attribute does not specify a collection, the Nullable attribute defaults to true. **The value true (or the absence of the Nullable attribute) indicates that no navigation target is required**. **The value false indicates that a navigation target is required** for the navigation property on instances of the containing type.

So we should not simply ignore whether navigation target is present but depend on the `Nullable` value of the navigation property to decide. Would you please update your pull request according to that? Thanks.

File: test/FunctionalTests/Tests/DataServices/UnitTests/ServerUnitTests1/Tests/ActionsTests.cs [hunk 2]
```diff
@@ -2022,7 +2022,7 @@ public void ActionResolveOrderTests()
                 {
                     RequestUri = "/Foo(1)/Bar",
                     Payload = "{ \"ID\" : 100 }",
-                    ExpectedResponse = "{\"@odata.context\":\"http://host/$metadata#Foo/$entity\",\"ID\":100}",
+                    ExpectedResponse = "{\"@odata.context\":\"http://host/$metadata#Foo(1)/Bar/$entity\",\"ID\":100}",
```
Comment (congysu) [inline test/FunctionalTests/Tests/DataServices/UnitTests/ServerUnitTests1/Tests/ActionsTests.cs]: The correct context url should be: {type-name} for an entity with optional navigation target.

**Comment by congysu** [comment]
Add to design backlog.

**Comment by lewischeng-ms** [comment]
Hi @sabyabr This issue has been fixed in  https://github.com/OData/odata.net/commit/74969d976eb913fa6ce30cd992943750af6f1cb2. Please see the following test case for reference: https://github.com/OData/odata.net/blob/master/test/FunctionalTests/Tests/DataOData/Tests/OData.TDD.Tests/Writer/JsonLight/ODataJsonLightFeedWriterTests.cs. We will close this pull request now but thank you all the same for your contribution.

**Pull request closed without merging**